api/v1/common/sagemaker_api.go

```diff
@@ -528,3 +528,45 @@ type DebugRuleEvaluationStatus struct {
 
 	StatusDetail *string `json:"statusDetail,omitempty"`
 }
+
+// AutoscalingResource TODO: <link>, check validation
+type AutoscalingResource struct {
+	// +kubebuilder:validation:MinLength=1
+	EndpointName *string `json:"endpointName"`
+
+	// +kubebuilder:validation:MinLength=1
+	VariantName *string `json:"variantName"`
+}
+
+// PredefinedMetricSpecification TODO: <link>
+type PredefinedMetricSpecification struct {
+	PredefinedMetricType *string `json:"predefinedMetricType,omitempty"`
+}
+
+// CustomizedMetricSpecification TODO: <link>
+type CustomizedMetricSpecification struct {
+	MetricName *string `json:"metricName,omitempty"`
+	Namespace  *string `json:"namespace,omitempty"`
+	Statistic  *string `json:"statistic,omitempty"`
+	Unit       *string `json:"unit,omitempty"`
+	// TODO: currently reusing the existing struct but this could be changed to match the API
+	Dimensions []AutoscalingResource `json:"dimensions,omitempty"`
+}
+
+// TargetTrackingScalingPolicyConfig TODO: <link>
+type TargetTrackingScalingPolicyConfig struct {
+	TargetValue      *int64 `json:"targetValue,omitempty"`
+	ScaleInCooldown  *int64 `json:"scaleInCooldown,omitempty"`
+	ScaleOutCooldown *int64 `json:"scaleOutCooldown,omitempty"`
+
+	// TODO: only one of the following two should be specified, wrap into another struct ?
```

Comment:
I think you still need to make them separate here in structure.

Reply:
Yea, makes sense